### PR TITLE
Projects from main repo

### PR DESCRIPTION
**Pull request by ivangrek**
ref https://github.com/gitextensions/gitextensions/pull/8574

- GitExtensions.Core split from main repo
- GitExtensions.Extensibility split from main repo

### CODE REVIEW

File: Directory.Build.targets [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (ivangrek) [reply Directory.Build.targets:1]: Without this, the CI build does not pass.

```
<!-- Create a dummy test targets -->
<Target Name="Test" />
<Target Name="IntegrationTest" />
<Target Name="LegacyClean" />
```

File: src/GitExtensions.Core/Commands/GitCommandConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            Default = new GitCommandConfiguration();
+
+            Default.Add(new GitConfigItem("rebase.autoSquash", "false"), "rebase");
+
+            Default.Add(new GitConfigItem("log.showSignature", "false"), "log", "show", "whatchanged");
+
+            Default.Add(new GitConfigItem("diff.submodule", "short"), "diff");
+            Default.Add(new GitConfigItem("diff.noprefix", "false"), "diff");
+            Default.Add(new GitConfigItem("diff.mnemonicprefix", "false"), "diff");
+            Default.Add(new GitConfigItem("diff.ignoreSubmodules", "none"), "diff", "status");
```
[the remark this answers is not in the document]
Comment (ivangrek) [reply src/GitExtensions.Core/Commands/GitCommandConfiguration.cs:30]: GitArgumentBuilder dependent on.

`public static GitCommandConfiguration Default { get; }`

`commandConfiguration = commandConfiguration ?? GitCommandConfiguration.Default;`

If there is a way to invert the dependency, I'd love to do it.

File: src/GitExtensions.Extensibility/RepositoryHosts/IPasswordHelper.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace GitExtensions.Extensibility.RepositoryHosts
+{
+    public interface IPasswordHelper
```
[the remark this answers is not in the document]
Comment (ivangrek) [reply src/GitExtensions.Extensibility/RepositoryHosts/IPasswordHelper.cs:3]: No.

File: src/GitExtensions.Core/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+
+[assembly: AssemblyDescription("GitExtensions core")]
```
[the remark this answers is not in the document]
Comment (ivangrek) [reply src/GitExtensions.Core/Properties/AssemblyInfo.cs:4]: How about this?

`GitExtensions core library. Contains base classes and interfaces.`

File: src/GitExtensions.Core/Properties/AssemblyInfo.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+[assembly: InternalsVisibleTo("GitExtUtils.Tests")]
+[assembly: InternalsVisibleTo("GitCommands.Tests")]
+[assembly: InternalsVisibleTo("GitExtensions")]
+[assembly: InternalsVisibleTo("CommonTestUtils")]
+[assembly: InternalsVisibleTo("TranslationApp")]
```
[the remark this answers is not in the document]
Comment (ivangrek) [reply src/GitExtensions.Core/Properties/AssemblyInfo.cs:10]: Allow access for

image

`ThreadHelper.JoinableTaskContext = new JoinableTaskContext();`

File: src/GitExtensions.Core/Commands/GitCommandConfiguration.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (RussKie) [reply src/GitExtensions.Core/Commands/GitCommandConfiguration.cs:30]: I meant moving population of `Default` values off to the app. Having these here would likely make it very opaque, i.e. _where do the Default values come from?_

File: Directory.Build.targets (hunk 1 above)
[the remark this answers is not in the document]
Comment (RussKie) [reply Directory.Build.targets:1]: This project should be build and distributed as a NuGet package... So it wouldn't rely on the Arcade infra.
If you're building it as part of the main solution - yes, we'd need this.

[the remark this answers is not in the document]
Comment (ivangrek) [reply Directory.Build.targets:1]: Currently connected as a submodule.

File: src/GitExtensions.Core/Properties/AssemblyInfo.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (RussKie) [reply src/GitExtensions.Core/Properties/AssemblyInfo.cs:10]: Just make it public.

File: src/GitExtensions.Core/Properties/AssemblyInfo.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (RussKie) [reply src/GitExtensions.Core/Properties/AssemblyInfo.cs:4]: Maybe `Core library containing interfaces and base implementations`?

File: src/GitExtensions.Extensibility/RepositoryHosts/IPasswordHelper.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (RussKie) [reply src/GitExtensions.Extensibility/RepositoryHosts/IPasswordHelper.cs:3]: Nuke it then :)

File: src/GitExtensions.Core/Commands/GitCommandConfiguration.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (gerhardol) [reply src/GitExtensions.Core/Commands/GitCommandConfiguration.cs:30]: The Git config options are related to the info/parsing of the commands and the dependency will be weird if the defaults are separated from the parsing.

**Comment by RussKie** [comment]
Is this PR migrating the history as well?

File: src/GitExtensions.Core/Commands/GitCommandConfiguration.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ivangrek) [reply src/GitExtensions.Core/Commands/GitCommandConfiguration.cs:30]: Need help finding a place to do this. Now it happens in the static constructor.

File: src/GitExtensions.Core/Properties/AssemblyInfo.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (ivangrek) [reply src/GitExtensions.Core/Properties/AssemblyInfo.cs:4]: Ok. I just see that the description of projects starts with GitExtensions.

**Comment by ivangrek** [comment]
> Is this PR migrating the history as well?

As I understand it, yes.

This seems like the best it can get, but I'm not sure.

image

File: src/GitExtensions.Extensibility/RepositoryHosts/IPasswordHelper.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (ivangrek) [reply src/GitExtensions.Extensibility/RepositoryHosts/IPasswordHelper.cs:3]: Ok.

File: src/GitExtensions.Core/GitExtensions.Core.csproj [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<Project Sdk="Microsoft.NET.Sdk">
+    <PropertyGroup>
+        <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    </PropertyGroup>
+
+    <ItemGroup>
+        <PackageReference Include="Ben.Demystifier" Version="0.1.4" />
+        <PackageReference Include="JetBrains.Annotations" Version="2018.2.1" />
```
[the remark this answers is not in the document]
Comment (ivangrek) [reply src/GitExtensions.Core/GitExtensions.Core.csproj:8]: What to do with such attributes?

```
[ContractAnnotation("=>false,objectId:null")]
[ContractAnnotation("=>true,objectId:notnull")]
```

**Pull request closed without merging**